src/setEnvVars.ts

```diff
@@ -76,6 +76,8 @@ export function setEnvVars({
   const { env: envVarsFromConfig } = getNextConfig();
   // Runtime and dotfile env vars are scoped by environment (via NEXT_PUBLIC_ prefix),
   // while env vars coming from next.config.js are available in both environments
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore NODE_ENV expected, but we are passing it so its okay
```

Comment:
if we really have to skip type checking we can do it with a `// @ts-expect-error {description}`. But it would be better if we could remove `src/next-env.d.ts`, if possible.

Reply:
Or.. another strategy: couldn't we just declare `NODE_ENV` prop :)? (I know we already provide it implicitly)

```ts
  process.env = {
    NODE_ENV: process.env.NODE_ENV,
    ...getEnvVarsByEnvironment(baseEnvVars)[runtimeEnv],
    ...envVarsFromConfig,
  };
```

Reply:
Or (2) we might make our `EnvVars` type more specific to extend `ProcessEnv`?